### PR TITLE
SMD-MDB-CF - Check Find model soundness

### PR DESCRIPTION
**Pull request by VLADISLAV008**

### CODE REVIEW

File: src/main/scala/group/scala/karazin/mongo4s/model.scala [hunk 1]
```diff
@@ -120,17 +120,20 @@ object model:
                              ordered: Option[Boolean] = None,
                              writeConcern: Option[WriteConcern] = None) derives Codec.AsObject
 
-  final case class Find[Filter, Sort, Projection](find: String,
+  final case class Find[Filter, Sort, Projection, Hint, Min, Max](find: String,
                                                   filter: Option[Filter] = None,
                                                   sort: Option[Sort] = None,
                                                   projection: Option[Projection] = None,
+                                                  hint: Option[Either[String, Hint]] = None,
```
Comment (mikhailZakharevich) [inline src/main/scala/group/scala/karazin/mongo4s/model.scala]: I think we could use Union type here (Option[String | Hint])

**Review by mikhailZakharevich — CHANGES_REQUESTED** [review]

File: src/main/scala/group/scala/karazin/mongo4s/model.scala [hunk 2]
```diff
@@ -124,12 +124,12 @@ object model:
...
                                                   sort: Option[Sort] = None,
                                                   projection: Option[Projection] = None,
-                                                  hint: Option[Either[String, Hint]] = None,
+                                                  hint: Option[Hint] = None,
                                                   skip: Option[Int] = None,
                                                   limit: Option[Int] = None,
                                                   batchSize: Option[Int] = None,
                                                   singleBatch: Option[Boolean] = None,
-                                                  comment: Option[Any] = None,
+                                                  comment: Option[JsonObject] = None,
```
Comment (mikhailZakharevich) [inline src/main/scala/group/scala/karazin/mongo4s/model.scala]: from the docs:
`A comment can be any valid BSON type (string, integer, object, array, etc).`

**Review by mikhailZakharevich — CHANGES_REQUESTED** [review]

File: src/main/scala/group/scala/karazin/mongo4s/model.scala (hunk 2 above)
Comment (VLADISLAV008) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala]: Good point. I missed the point that JsonObject does not suit all of these types.

File: src/main/scala/group/scala/karazin/mongo4s/model.scala [hunk 3]
```diff
@@ -129,7 +129,7 @@ object model:
                                                   limit: Option[Int] = None,
                                                   batchSize: Option[Int] = None,
                                                   singleBatch: Option[Boolean] = None,
-                                                  comment: Option[JsonObject] = None,
+                                                  comment: Option[Json] = None,
```
Comment (mikhailZakharevich) [inline src/main/scala/group/scala/karazin/mongo4s/model.scala]: Yeah, that's better. :)
Although, you could even make a separate type parameter for that to let people decide what it is.

Comment (VLADISLAV008) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala]: Something like this ` type BSON = Json`?

Comment (mikhailZakharevich) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala]: No, I mean you could do something like Find[Filter, Sort, Projection, Comment](..., comment: Option[Comment])

Comment (mikhailZakharevich) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala]: Because we eventually gonna parse literals/objects/collections to json

**Review by mikhailZakharevich — APPROVED** [review]
LGTM

**Pull request merged**